app/ante/utils_test.go

```diff
@@ -343,6 +345,52 @@ func (suite *AnteTestSuite) CreateTestEIP712MsgSubmitEvidence(from sdk.AccAddres
 	return suite.CreateTestEIP712CosmosTxBuilder(from, priv, chainId, gas, gasAmount, msgEvidence)
 }
 
+func (suite *AnteTestSuite) CreateTestEIP712SubmitProposalV1(from sdk.AccAddress, priv cryptotypes.PrivKey, chainId string, gas uint64, gasAmount sdk.Coins) client.TxBuilder {
+	// Build V1 proposal messages. Must all be same-type, since EIP-712
+	// does not support arrays of variable type.
+	authAcc := suite.app.GovKeeper.GetGovernanceAccount(suite.ctx)
+
+	proposal1, ok := types5.ContentFromProposalType("My proposal 1", "My description 1", types5.ProposalTypeText)
```

Comment:
nit, these `typesN` are a bad pattern. We should rename them to `govtypes`, `banktypes`, etc

Reply:
Personally agreed, although in this case, I'm using existing import definitions. Think we should use a separate PR to refactor.